pkg/discovery/discovery.go

```diff
@@ -35,17 +35,17 @@ import (
 )
 
 // Run is the main entrypoint for discovery
-func Run(kubeClient kubernetes.Interface, cfg *config.Config) (errCount uint) {
+func Run(kubeClient kubernetes.Interface, cfg *config.Config) (errCount uint, err error) {
 	t := time.Now()
 
 	// 1. Create the directory which will store the results, including the
 	// `meta` directory inside it (which we always need regardless of
 	// config)
 	outpath := path.Join(cfg.ResultsDir, cfg.UUID)
 	metapath := path.Join(outpath, MetaLocation)
-	err := os.MkdirAll(metapath, 0755)
+	err = os.MkdirAll(metapath, 0755)
 	if err != nil {
-		panic(err.Error())
+		return errCount, errors.Wrap(err, "could not create directory to store results")
```

Comment:
So a while ago we used to pass all the errors back to main, but it became weird and counterproductive, because it was always a list of errors and you lost context from where it occurred in the query stream.  

logging the error in place and returning for graceful exit seems sufficient and keeps the funcion consistent.   Otherwise for consistency, we'd have to plumb back the error[] from the queries.

Reply:
Ahh I see. So instead of returning the error, Run should:
* Continue to return a single value:`errCount`
* Log the error, increment the count, and return early for graceful exit

Is my understanding correct? Thanks!

Reply:
@alexbrand - yup!